syntax = "proto3";


// The greeting service definition.
service MessageSync {
  // message producer
  rpc SubmitMessage (SubmitRequest) returns (SubmitReply) {}

  // message consume, push mode
  rpc PushMessageStream(ConnRequest) returns (stream MessageReply){}
}

// The request message containing the user's name.
message SubmitRequest {
  string channel = 1;
  string message = 2;
}

message SubmitReply {
  string reply = 1;
}

// The response message containing the greetings
message ConnRequest {
  string channel = 1;
}

message MessageReply{
  string message = 1;
}